kyo-prelude/shared/src/main/scala/kyo2/Abort.scala

```diff
@@ -0,0 +1,93 @@
+package kyo2
+
+import kernel.<
+import kernel.Const
+import kernel.Effect
+import kernel.Frame
+import kyo.Tag
+import scala.reflect.ClassTag
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+sealed trait Abort[-E] extends Effect[Const[E], Const[Unit]]
+
+object Abort:
+
+    private inline def erasedTag[E]: Tag[Abort[E]] = Tag[Abort[Any]].asInstanceOf[Tag[Abort[E]]]
```

Comment:
We still need untyped effect handling here because it's not possible to use `<:<` to compare tags or use `Tag.Full` in the kernel due to the performance cost. I'm planning to continue the work to integrate union and intersection type tags into the tag string representation itself and avoid string scans by prefixing payloads with their length, which should enable us to avoid the erased tag here and in `Env` (see branch https://github.com/getkyo/kyo/tree/tags-take-3)